tests/test_docstring_checker.py

```diff
@@ -0,0 +1,31 @@
+import logging
+import os
+
+from docs import check_docstrings
+
+
+def test_check_blank_line(caplog):
+    test_string = os.path.join(
+        os.path.dirname(__file__), "saved_test_data", "sample_docstrings.py"
+    )
+
+    caplog.clear()
+    caplog.set_level(logging.ERROR)
+    error_count = check_docstrings.check_blank_line_above_param_section(test_string)
+
+    # Line numbers of good and bad docstrings in sample_docstrings.py
+    good_doc_line_nums = [2, 16, 25, 35]
+    bad_doc_line_nums = [43, 53, 65]
+
+    # Check that good docstrings do not log error
+    for line_num in good_doc_line_nums:
+        msg = f"sample_docstrings.py: {line_num}: Must have exactly 1 blank line"
+        assert msg not in caplog.text
+
+    # Check that bad docstrings log error
+    for line_num in bad_doc_line_nums:
```

Comment:
Not worth refactoring at this point, just a comment.

The logic here could be much simpler here (probably more than half of this could be rm'd). The bad_nums, and only the bad_nums should be captured.  Anything else of this form would an error and could just raise/report, no need to loop/compare/etc.

Reply:
Good point. Will keep in mind for future.